### PR TITLE
Rename TCK "ant-based" to "platform"

### PR DESCRIPTION
**Pull request by arjantijms**

### CODE REVIEW

File: appserver/tests/tck/tck-download/jakarta-platform-tck/pom.xml [hunk 1]
```diff
@@ -27,14 +27,14 @@
...
+    <artifactId>jakarta-platform-tck</artifactId>
     <packaging>pom</packaging>
-    <name>TCK: Install Jakarta EE TCK</name>
+    <name>TCK: Install Jakarta EE Platform TCK</name>
 
     <properties>
         <!-- Be patient, it is over 500 MB -->
         <tck.test.jee.file>jakarta-jakartaeetck-10.0.2.zip</tck.test.jee.file>
-        <tck.test.jee.url>https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee10/staged/eftl/${tck.test.jee.file}</tck.test.jee.url>
+        <tck.test.jee.url>https://download.eclipse.org/jakartaee/platform/10/${tck.test.jee.file}</tck.test.jee.url>
```
Comment (hs536) [inline appserver/tests/tck/tck-download/jakarta-platform-tck/pom.xml:37]: I'm not familiar with TCK prj, but would it be a problem to use a non-EFTL TCK?

Comment (hs536) [reply appserver/tests/tck/tck-download/jakarta-platform-tck/pom.xml:37]: Sorry, my mistake, it seems that after the final release of TCK it is correct to download from that URL.

**Review by hs536 — APPROVED** [review]

**Review by dmatej — CHANGES_REQUESTED** [review]
Compilation failed for some reason.
`/repo/git/glassfish/appserver/tests/tck/platform-tck-runner/src/main/java/org/glassfish/main/tests/tck/ant/TckRunner.java:[30,56] package org.glassfish.main.tests.tck.ant.junit.generated does not exist`

EDIT: Seems it is not your mistake, but mine, I will commit a fix to your PR ...
EDIT2: It was broken since https://github.com/eclipse-ee4j/glassfish/pull/24661 and I noticed something related first in #24616, so I cherrypicked the fix here now. The trick is that with the previous gfbuild version the compiler settings were ignored/split to two executions, but now they are not and the mapping was incorrect now.

**Review by dmatej — APPROVED** [review]
```
mvn clean install -Ptck -pl :jakarta-platform-tck
mvn clean install -Ptck -pl :platform-tck-runner -Dit.test=CustomITest -Dtck.module=signaturetest/javaee
```
```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  19:48 min
[INFO] Finished at: 2023-11-06T19:50:44+01:00
[INFO] ------------------------------------------------------------------------
```
JUnit report:
```
<testsuite  id="1" name="signaturetest_javaee" hostname="dmatej-P7740" tests="4" failures="0" errors="0" disabled="0" skipped="0" time="562" timestamp="2023-11-06T19:49:59">
```

**Pull request merged**